File: app/src/main/java/quintype/com/templatecollectionwithrx/models/collection/CoverImage.java
package quintype.com.templatecollectionwithrx.models.collection;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by akshaykoul on 18/04/16.
 */
public class CoverImage implements Parcelable {

    @SerializedName("cover-image-url")
    @Expose
    private String coverImageUrl;
    @SerializedName("cover-image-metadata")
    @Expose
    private CoverImageMetadata coverImageMetadata;
    @SerializedName("caption")
    @Expose
    private String caption;
    @SerializedName("cover-image-s3-key")
    @Expose
    private String coverImageS3Key;

    public static final Creator<CoverImage> CREATOR = new Creator<CoverImage>() {
        @Override
        public CoverImage createFromParcel(Parcel in) {
            return new CoverImage(in);
        }

        @Override
        public CoverImage[] newArray(int size) {
            return new CoverImage[size];
        }
    };

    public String coverImageUrl() {
        return coverImageUrl;
    }

    public CoverImageMetadata coverImageMetadata() {
        return coverImageMetadata;
    }

    public String caption() {
        return caption;
    }

    public String coverImageS3Key() {
        return coverImageS3Key;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.coverImageUrl);
        dest.writeParcelable(this.coverImageMetadata, flags);
        dest.writeString(this.caption);
        dest.writeString(this.coverImageS3Key);
    }

    public CoverImage() {
    }

    protected CoverImage(Parcel in) {
        this.coverImageUrl = in.readString();
        this.coverImageMetadata = in.readParcelable(CoverImageMetadata.class.getClassLoader());
        this.caption = in.readString();
        this.coverImageS3Key = in.readString();
    }
}

File: app/src/main/java/quintype/com/templatecollectionwithrx/models/collection/CoverImage.kt
package quintype.com.templatecollectionwithrx.models.collection

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 8/31/18.
 */

class CoverImage : Parcelable {

    @SerializedName("cover-image-url")
    @Expose
    private val coverImageUrl: String
    @SerializedName("cover-image-metadata")
    @Expose
    private val coverImageMetadata: CoverImageMetadata
    @SerializedName("caption")
    @Expose
    private val caption: String
    @SerializedName("cover-image-s3-key")
    @Expose
    private val coverImageS3Key: String

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.coverImageUrl)
        dest.writeParcelable(this.coverImageMetadata, flags)
        dest.writeString(this.caption)
        dest.writeString(this.coverImageS3Key)
    }

    protected constructor(parcel: Parcel) {
        this.coverImageUrl = parcel.readString()
        this.coverImageMetadata = parcel.readParcelable(CoverImageMetadata::class.java.classLoader)
        this.caption = parcel.readString()
        this.coverImageS3Key = parcel.readString()
    }

    companion object CREATOR : Parcelable.Creator<CoverImage> {
        override fun createFromParcel(parcel: Parcel): CoverImage {
            return CoverImage(parcel)
        }

        override fun newArray(size: Int): Array<CoverImage?> {
            return arrayOfNulls(size)
        }
    }
}
